starport-core/src/main/scala/com/krux/starport/metric/MetricSettings.scala

```diff
@@ -3,38 +3,107 @@ package com.krux.starport.metric
+      case CloudWatchReporterSettings => CloudWatchReporterSettings(config).getReporter(registry)
+  }
 
-  def getReporter(registry: MetricRegistry): ScheduledReporter = GraphiteReporter.forRegistry(registry)
-    .prefixedWith(metricPrefix)
-    .convertDurationsTo(MetricSettings.DefaultDuration)
-    .build(getGraphite)
+  def getDefaultReporter(registry: MetricRegistry): ScheduledReporter = ConsoleReporter
+    .forRegistry(registry)
+    .convertDurationsTo(DefaultDuration)
+    .build()
 
 }
 
-object MetricSettings {
+final case object GraphiteReporterSettings extends MetricSettings {
 
-  final val DefaultDuration = TimeUnit.SECONDS
+  def apply(config: Config): MetricSettingsImpl = GraphiteReporterSettingsImpl(config)
+
+  private case class GraphiteReporterSettingsImpl(config: Config) extends MetricSettingsImpl {
+
+    val hosts = config.getStringList("hosts")
+
+    val port = config.getInt("port")
+
+    val metricPrefix = config.getString("prefix")
+
+    def getGraphite: Graphite =
+      new Graphite(hosts.get(Random.nextInt(hosts.size)), port)
+
+    def getReporter(registry: MetricRegistry): ScheduledReporter = GraphiteReporter
+      .forRegistry(registry)
+      .prefixedWith(metricPrefix)
+      .convertDurationsTo(DefaultDuration)
+      .build(getGraphite)
+
+  }
 
-  def getReporter(configOpt: Option[Config], registry: MetricRegistry): ScheduledReporter =
-    configOpt.map(new MetricSettings(_).getReporter(registry))
-      .getOrElse(
-        ConsoleReporter.forRegistry(registry).convertDurationsTo(DefaultDuration).build()
-      )
 }
+
+final case object CloudWatchReporterSettings extends MetricSettings {
+
+  def apply(config: Config): MetricSettingsImpl = CloudWatchReporterSettingsImpl(config)
+
+  private case class CloudWatchReporterSettingsImpl(config: Config) extends MetricSettingsImpl {
+
+    val awsRegion: Region = config.getString("region").asInstanceOf[Region]
+    val awsEnvironment = config.getString("environment")
+
+    val awsCloudWatchAsync: CloudWatchAsyncClient = CloudWatchAsyncClient
+      .builder
+      .region(awsRegion)
+      .build
+
+    def getReporter(metricRegistry: MetricRegistry): CloudWatchReporter = CloudWatchReporter
+      .forRegistry(metricRegistry, awsCloudWatchAsync, classOf[Nothing].getName)
+      .convertRatesTo(TimeUnit.SECONDS).convertDurationsTo(TimeUnit.MILLISECONDS)
+      .filter(MetricFilter.ALL)
+      .withPercentiles(Percentile.P75, Percentile.P99)
+      .withOneMinuteMeanRate
+      .withFiveMinuteMeanRate
+      .withFifteenMinuteMeanRate
+      .withMeanRate
+      .withArithmeticMean
+      .withStdDev
+      .withStatisticSet
+      .withZeroValuesSubmission
+      .withReportRawCountValue
+      .withHighResolution
+      .withMeterUnitSentToCW(StandardUnit.BYTES)
+      .withJvmMetrics
+      .withGlobalDimensions(s"Region=${awsRegion}", s"Instance=${awsEnvironment}")
```

Comment:
👍

Reply:
Let's use `krux.starport.prefix` for this

Reply:
OK good call.